### PR TITLE
Adding Azure unittests

### PR DESCRIPTION
**Pull request by shahidee44**
## Description

- Adding Missing unittests for Azure Backend
- Added QiskitQPUBackend with Azure Device Exception messages check unittest
- Added remote qubit overflow with Azure Device unittest
- Added Manual Mode check for Azure Backend.
- Updated Old unittest with Azure backend
- Changed Exception Messages in Qiskit Backend

## Checklist

[//]: <> (- [ ] My code follows the style guidelines of this project)

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code and used numpy-style docstrings
- [ ] I have made corresponding updates to the documentation.
- [ ] My changes generate no new warnings
- [ ] I have added/updated tests to make sure bugfix/feature works.
- [ ] New and existing unit tests pass locally with my changes.

[//]: <> (- [ ] Any dependent changes have been merged and published in downstream modules)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Name the new unit-tests that you have added along with this change.

### CODE REVIEW

File: src/openqaoa-azure/backends/devices.py [hunk 1]
```diff
@@ -28,9 +28,9 @@ def __init__(self, device_name: str, resource_id: str, az_location: str):
         device_name: `str`
             The name of the Azure remote QPU/Simulator to be used
         resource_id: `str`
-            resource_id
+            The resource_id of the Workplace
         az_location: `str`
-            az_location
+            The location of the Azure Workplace. e.g. "us-west-1"
```
Comment (Q-lds) [inline src/openqaoa-azure/backends/devices.py:33]: Are you sure they write the locations in that way? I have only seen it like `eastus`

File: tests/test_qpu_qiskit.py [hunk 2]
```diff
@@ -21,17 +24,140 @@
...
+        
+        bashCommand = "az resource list"
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        
+        if error is not None:
+            print(error)
+            raise Exception('You must have the Azure CLI installed and must be logged in to use the Azure Quantum Backends')
+        else:
+            output_json = json.loads(output)
```
Comment (Q-lds) [inline tests/test_qpu_qiskit.py:50]: @shahidee44 have you already created a resource for this?

Comment (shahidee44) [reply tests/test_qpu_qiskit.py:50]: No. Currently we're just using the TestingOpenQAOA Workspace for tests.

File: src/openqaoa-azure/backends/devices.py (hunk 1 above)
Comment (shahidee44) [reply src/openqaoa-azure/backends/devices.py:33]: You're right. The location example here is from braket. ill change that

**Pull request merged**